### PR TITLE
chore: storybook upgrade

### PR DESCRIPTION
**Pull request by sharkparty**
- Upgraded repo to storybook 5
- Upgraded typescript
- Upgraded types
- Upgraded jest
- Upgraded Button to fix type bug exposed with new version of TS/SC

**Before submitting a pull request,** please make sure the following is done:

- [x] Clone [the repository](https://github.com/retailmenot/anchor) and branch from `master` (`git checkout -b feature/button`).
- [x] Run `npm ci` in the repository root.
- [x] Use [conventional commits](https://www.conventionalcoamits.org/en/v1.0.0-beta.3/) for all work.
- [x] If you've fixed a bug or added code, [add stories](https://storybook.js.org/docs/basics/introduction/)!
- [x] If you've fixed a bug or added code that should be tested, [add tests](https://jestjs.io/docs/en/getting-started)!
- [x] Ensure the test suite passes & coverage thresholds are met (`npm test`).
- [x] Format your code with [prettier](https://github.com/prettier/prettier) (`npm run pretty`).
- [x] Make sure your code lints (`npm run lint`).

---------
**Outline your feature or bug-fix below**

This PR upgrades this repo to the latest version of storybook and fixes some typing issues.

### CODE REVIEW

**Comment by bjohnsonrmn** [comment]
@sharkparty, I see a bunch of files in `docs` that are removed in this commit. Was that intentional?

File: package.json [hunk 1]
```diff
@@ -121,7 +124,7 @@
     "rxjs": "^6.3.3"
   },
   "engines": {
-    "node": ">=8"
+    "node": ">=10"
```
Comment (bjohnsonrmn) [inline package.json:127]: @sharkparty could this be where the `0.10.x` is coming from?

Comment (sharkparty) [reply package.json:127]: I bumped this back up and defined the version of node in the travis yaml file. Idiot me didn't think that travis would default to a prehistoric version of node. Surprise surprise. I also had deleted the docs directory to do some testing and didn't bring that back, so I added it back. All is good now.

**Comment by bjohnsonrmn** [comment]
@sharkparty looks good to me now.

**Comment by sharkparty** [comment]
Worth noting: I did a clean install with both yarn and npm to make sure there wasn't any dependency anomalies and I didn't see any during runtime or developer testing, but the trees are subtly different. If anyone has a specific issue, just open an issue and I'll get to it.

**Pull request merged**